### ISSUE TITLE
Could not get kernel from Jupyter extension. This likely means that the Data Wrangler extension is not compatible with the Jupyter extension. Please file an issue on the Data Wrangler GitHub repository.

### ISSUE THREAD

**Issue by onceone** [microsoft/vscode-data-wrangler#254]
**Could not get kernel from Jupyter extension. This likely means that the Data Wrangler extension is not compatible with the Jupyter extension. Please file an issue on the Data Wrangler GitHub repository.**
## Environment data

-   VS Code version: XXX
-   Data Wrangler Extension version (available under the Extensions sidebar): XXX
-   Jupyter Extension version (available under the Extensions sidebar): XXX
-   Python Extension version (available under the Extensions sidebar): XXX
-   OS (Windows | Mac | Linux distro) and version: XXX
-   Pandas version: XXX
-   Python and/or Anaconda version: XXX
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): XXX

## Expected behaviour

XXX

## Actual behaviour

XXX

## Steps to reproduce:

[**NOTE**: Self-contained, minimal reproducing code samples are **extremely** helpful and will expedite addressing your issue]

1. XXX

## Logs

<details>

<summary>Output for <code>Jupyter</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Jupyter</code>)
</summary>

<p>

```
XXX
```

</p>
</details>

**Comment by pwang347** [comment]
Hi @onceone, thanks for opening this issue! Could you please share more information about the version of Jupyter extension and Data Wrangler extension you have are?

Are you able to reproduce this on the latest stable version for both extensions? Thanks!

**Comment by BangyStudios** [comment]
This issue isn't with Jupyter Notebooks, but with the VSCode Python debugger.

Since VSCode removed the built-in data viewer from the debugger, there's no way to view data, such as a pandas DataFrame, in memory within the debugger without using a third-party tool.

This is the extension you're redirected to when you attempt to inspect a variable inside the Python debugger.

![Screenshot 2024-08-12 at 10 46 45](https://github.com/user-attachments/assets/247769b4-3b58-4ee4-b64c-d0539db401d5)

![Screenshot 2024-08-12 at 10 48 13](https://github.com/user-attachments/assets/cd1229bc-00ba-4f56-b494-765176f6c446)

![Screenshot 2024-08-12 at 10 48 35](https://github.com/user-attachments/assets/2973169a-1ab4-46bb-bdeb-b64f509fa500)

**Comment by elsha-phytech** [comment]
**I'm experiencing the same issue with Data Wrangler not working in Jupyter.**

**Environment:**
- **Jupyter Pack:** v2024.7.0
- **Data Wrangler:** v1.6.0

### Details

I installed Data Wrangler to view data while debugging (since it's the only third-party tool shown as an option after the removal of the previous feature). However, when I try to view any data, I encounter the following error:

> **Error Message:**  
> Could not get kernel service from Jupyter extension. This likely means that the Data Wrangler extension is not compatible with the Jupyter extension. Please file an issue on the Data Wrangler GitHub repository.
> 
> **Source:** Data Wrangler

I am using the latest stable versions of both the Jupyter pack and Data Wrangler.


Thank you for your support! 🙏

**Comment by pwang347** [comment]
@elsha-phytech Hmm that is weird, I can't seem to reproduce this issue with the same versions.

Could you please share the following info:
1. How are you trying to launch the viewer? Is it a .PY file or .IPYNB? If using .PY, are you using the [interactive window](https://code.visualstudio.com/docs/python/jupyter-support-py) feature or just local debugger?
2. Are you able to view data from the following toolbar item in a .IPYNB file?
<img width="314" alt="image" src="https://github.com/user-attachments/assets/9edea628-8c4a-4da7-b151-f0bb6ca1de6a">

3. Are you able to open a local CSV file using Data Wrangler (e.g. https://raw.githubusercontent.com/pwang347/cdn/master/AB_NYC_2019.csv) when selecting the local Python interpreter option?
<img width="785" alt="image" src="https://github.com/user-attachments/assets/44b5e35c-b67f-4124-92e5-f5c2f1dc934a">

4. Could you please set the Jupyter logging level to "debug" and then reproduce the issue again? You can find this setting by searching for "jupyter logging" from the settings UI.
<img width="561" alt="image" src="https://github.com/user-attachments/assets/0701e7e5-bf48-4e4b-a345-05e3daa696f0">

We would then appreciate if you could then copy and attach the logs in the Output panel here:
<img width="486" alt="image" src="https://github.com/user-attachments/assets/bd17ff2f-9c76-4a69-977a-ba92f95fc680">

Thank you!

**Comment by pwang347** [comment]
@BangyStudios Are you seeing the same error message as in the title? If not, could you please check if this thread resolves your issue? https://github.com/microsoft/vscode-data-wrangler/issues/253#issuecomment-2276876252

Otherwise, it would be great if you could also provide the information I've shared in the previous message to help us investigate further.

Thanks!

**Comment by bjunkerm** [comment]
@pwang347 I am experiencing the same problem as described. I went through your suggestions, here is the result
1) I am also trying to launch the viewer from the debugger (variables view) as descibed by @BangyStudios. I am debugging a unit test triggered via vs code lab.
2) Viewing a dataframe from a jupyter notebook using the toolbar works
3) Viewing a csv file with local interpreter works as well 
4) Changing the setting does not help
5) The log only showed: 08:45:05.889 [debug] Evaluating in debugger : e54010f3-ca26-46fb-830e-e703e8691ecd: _VSCODE_getVariable("info", True, sample_df)

I tried the suggestion from https://github.com/microsoft/vscode-data-wrangler/issues/253#issuecomment-2276876252 but this did not work either.

However I noticed when I checked 2) and launched a jupyter kernel, the error message changed to "Could not retrieve variable sample_df from the Jupyter extension. Please file an issue on the Data Wrangler GitHub repository."

So it seems to me that there is attempt to retrieve the value from jupyter when I am not even working with a jupyter notebook and instead: debugging a unit test

**Comment by elsha-phytech** [comment]
@pwang347
> *So I've spent the last 40 minutes to write a detailed answer, but apparantly github decided it was too much because suddenly everything I've written was gone.  
I'll try to rewrite everything and do it short and conscise.*

First of all, the issue was that I haven't installed the jupyter python package.
The solution that worked for me:
- make sure you are in the same conda env as the interpreter used for debugging
- ```conda install jupyter```

Secondly, if this is a requirement - it will be helpful to mention it on the extantion page (the only Jupyter req. mentioned at the 'Set up your environment' section is the VSC Jupyter extantion pack (which I had as said)).
Perhaps even consider writing it in a more noticeable manner, making it easier for those struggling with this issue to find.  
Another approach is to suggest to the user to install the missing package (as done by you for missing Pandas, or by Jupyter ext. when there is no ipykernel present in the chosen interpreter's env).
It's important especially since the use case of Data-Wrengler is not limited to Jupyter context (e.g. my case - debugging a general python run), so installing the jupyter package isn't a simple solution to come to mind. 

Anyway I will provide the info requested:  
1. I've tried to launch using the local debugger.
2. Yes (the View Data button exsits and functioning).
3. Yes (viewed csv successfully).
4. It was checked ('Experimental').
5. I fail to reproduce the issue now 🤷🏻‍♂  
If someone which hasn't yet fixed this can upload the logs please do 🙏🏻

Finally, now that it works it seems really nice.  
Thank you for all your work.

**Comment by Dr3ven** [comment]
## Environment data
VS Code version: 1.92.2
Data Wrangler Extension version (available under the Extensions sidebar): v1.6.0
Jupyter Extension version (available under the Extensions sidebar): v2024.7.0
Python Extension version (available under the Extensions sidebar): v2024.12.3
OS (Windows | Mac | Linux distro) and version: Debian 12 (bookworm)
Pandas version: 2.2.2
Python and/or Anaconda version: 3.11.2
Type of virtual environment used (N/A | venv | virtualenv | conda | ...): venv
## Expected behaviour
I wanted to look at an array and open it with DataWrangler while using the Python debugger. 

## Actual behaviour
When right-clicking on the array in the Python debugger tab and selecting `View Value in Data Viewer` the error from the issue title occurred: "Could not get kernel from Jupyter extension. This likely means that the Data Wrangler extension is not compatible with the Jupyter extension. Please file an issue on the Data Wrangler GitHub repository."

After googling the issue I ended up here and read the steps to up to the stage of @pwang347's (https://github.com/microsoft/vscode-data-wrangler/issues/254#issuecomment-2285122493). I  did not download the csv but still enabled the options from step `4.` and `5.` (`Used Jupyter proposed API` and `Jupyter Loggin Level: debug`). These action alone already fixed the error for me so that I could open that array with DataWrangler.
Just wanted to share my experience so that it may help others coming here.

**Comment by pwang347** [comment]
Hi all, thanks for the patience! Just wanted to update that we've released some changes in pre-release 1.7.2 that should potentially address the issue. Would love if someone could give it a try and let me know if they are still seeing the error message in the original post.

Thank you!

**Comment by bjunkerm** [comment]
Hi @pwang347  it now works for my case without any error messages with version 1.7.2. Thanks a lot for fixing this so quickly!

**Comment by pwang347** [comment]
@bjunkerm Awesome, thank you very much for confirming! I will leave this thread open for a bit longer in case anyone else is still running into the issue even on 1.7.2, but plan to close it in next few days if it's not reproducing anymore. Cheers!

**Comment by pwang347** [comment]
Hi @IwonaZwierzynska! I see that the error message is different from the one in the original issue, could you please file a new one and we can track that one separately?

I would really appreciate if you could include the following information in the post as well:
- How frequently is the issue happening for you? Is it all the time or only in some scenarios?
- The type of data you are launching (e.g. large Pandas DataFrame with strings and numbers, etc.)
- Are you running into the same issue even if it is something simple like the following?

```python
import pandas as pd
df = pd.DataFrame(data={'col1': [1, 2], 'col2': [3, 4]})`
```

- Any errors or abnormal messages in the developer console logs that seem related?
![image](https://github.com/user-attachments/assets/b938cb62-1a06-4f51-8278-8af6452624f9)
And then copy the text in the "Console" tab:
![image](https://github.com/user-attachments/assets/767fbefa-6e8c-4523-828d-07dc605f8002)
- Do you have multiple Python environments? If so, could you please check if the issue is occurring in other environments as well? You can change it by clicking the environment selector in the bottom right:
<img width="308" alt="image" src="https://github.com/user-attachments/assets/01ee2ef8-1cbc-4fc6-84bb-40bb03cff4df">
 
- Lastly, are you using the [interactive window](https://code.visualstudio.com/docs/python/jupyter-support-py) debugger or the default debugpy? ie. what button or command do you press to start debugging?

It would greatly help us with the investigation if you could provide this info, thank you!

**Comment by IwonaZwierzynska** [comment]
Thank you very much for your reply. The new issue is available under: https://github.com/microsoft/vscode-data-wrangler/issues/265

**Comment by NiKoenig** [comment]
Are there any updates on this issue? I am experiencing the same problem, both in the VSCode Interactive Window as well as the Debugger.

**Comment by pwang347** [comment]
Will close this issue since there haven't been any new comments (and the fix is now available in 1.8.0). Please feel free to re-open if this is still happening to you on the latest versions!

**Comment by mrjoe3012** [comment]
Recently had this issue 1.99.3, all extensions up to date at the time. Fixed the issue with @elsha-phytech suggestion `pip install jupyter` in the same environment as the kernel.

![Image](https://github.com/user-attachments/assets/87872875-d33c-418c-8dd3-781fbea1d8df)
